### PR TITLE
Converts tinymce to service providers

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the tinymce editor to service providers.

### Testing Instructions
- Make sure TinyMCE is the default editor
- Edit an article

Also test if the other editors do still work

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@laoneo thank you for the one! I'll test it in a bit

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 53b65edf57c6c45ed46fb173747452ec2d674c00<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

File: plugins/editors/tinymce/src/Field/TinymcebuilderField.php [hunk 1]
```diff
@@ -15,6 +15,7 @@
 use Joomla\CMS\Form\FormField;
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Plugin\Content\Contact\Extension\TinyMCE;
```
Comment (dgrammatiko) [inline plugins/editors/tinymce/src/Field/TinymcebuilderField.php]:
```suggestion
use Joomla\Plugin\Editors\TinyMCE\Extension\TinyMCE;
```

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by dgrammatiko** [comment]
I have tested this item :red_circle: unsuccessfully on 53b65edf57c6c45ed46fb173747452ec2d674c00<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

**Comment by laoneo** [comment]
I'v uploaded a fix, it should work now. Strange thing was that it loaded without that change for me

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 53b65edf57c6c45ed46fb173747452ec2d674c00<br /><br />Tip for the nest tester: delete the file `administrator/cache/autoload_psr4.php` prior testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 15e3454eb60342590bc7850b61fc36c492c0293c<br /><br />Based on the test instructions I am marking this as a successful test. Can't really comment on the code changes themselves<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 15e3454eb60342590bc7850b61fc36c492c0293c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

File: libraries/src/Editor/Editor.php [hunk 2]
```diff
@@ -277,38 +277,22 @@ protected function _loadEditor($config = array())
...
+        // Check if the editor can be loaded
+        if (!$this->_editor) {
+            Log::add(Text::_('JLIB_HTML_EDITOR_CANNOT_LOAD'), Log::WARNING, 'jerror');
 
-        // If no plugin is published we get an empty array and there not so much to do with it
-        if (empty($plugin)) {
             return false;
         }
 
-        $params = new Registry($plugin->params);
```
Comment (Quy) [inline libraries/src/Editor/Editor.php:299]: Remove import.

**Comment by laoneo** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39625">issues.joomla.org/tracker/joomla-cms/39625</a>.</sub>

**Comment by obuisard** [comment]
Thank you Allon @laoneo!

**Pull request merged**